src/OrchardCore/OrchardCore.ResourceManagement/ResourceManager.cs

```diff
@@ -206,9 +206,8 @@ private ResourceDefinition FindResource(RequireSettings settings, bool resolveIn
                             }
                         }
 
-                        // Use the highest version of all matches.
-                        if (resource == null
-                            || (resourceDefinition.Version != null && new Version(resource.Version) < version))
+                        // Use the highest version of all matches. Resources that don't specify a version have the lowest priority.
+                        if (resource == null || resource.Version == null || new Version(resource.Version) < version)
```

Comment:
Please simplify code to this 
```
resource?.Version is null
```

Reply:
@MikeAlhayek you could use the suggest feature in GitHub instead of writing the code yourself, it's quite helpful